File: collections/src/main/java/com/czmp/collections/dto/ItemDTO.java
package com.czmp.collections.dto;

import com.czmp.collections.model.Item;
import com.czmp.collections.model.ItemCollection;
import com.czmp.collections.model.Tag;
import lombok.Data;

import java.util.List;

@Data
public class ItemDTO {
    private Long id;
    private String name;
    private String description;
    private Item.Status status;
    private List<String> tags;
}

File: CollectionsAPP/src/app/shared/connection/models/item.dto.ts
import {EndUserDto} from "./end-user.dto";
import {Tag} from "./tag";
import {ItemCollectionDto} from "./item-collection.dto";

export class ItemDto {
  id: number = 0
  name: string = '';
  description: string = '';
  status: Status = Status.notForSale;
  tags: Tag[] = [];


  constructor(id: number, name: string, description: string, status: Status, tags: Tag[]) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.status = status;
    this.tags = tags;
  }
}

export enum Status {
  forSale = 'FOR_SALE',
  notForSale = 'NOT_FOR_SALE',
}
